cca_example_test.go

```diff
@@ -0,0 +1,187 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat_test
+
+import (
+	"fmt"
+	"log"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix/mat64"
+	"github.com/gonum/stat"
+)
+
+// Helper for getting a View of a SymDense.
+type symView struct {
+	sym        *mat64.SymDense
+	i, j, r, c int
+}
+
+func (s symView) Dims() (r, c int) { return s.r, s.c }
+
+func (s symView) At(i, j int) float64 {
+	if i < 0 || s.r <= i {
+		panic("i out of bounds")
+	}
+	if j < 0 || s.c <= j {
+		panic("j out of bounds")
+	}
+	return s.sym.At(s.i+i, s.j+j)
+}
+
+func (s symView) T() mat64.Matrix { return mat64.Transpose{s} }
+
+func ExampleCanonicalCorrelations() {
+	// This example is directly analogous to Example 3.5 on page 87 of
+	// Koch, Inge. Analysis of multivariate and high-dimensional data.
+	// Vol. 32. Cambridge University Press, 2013. ISBN: 9780521887939
+
+	// BostonData is the Boston Housing Data of Harrison and Rubinfeld (1978)
+	n, _ := stat.BostonData.Dims()
```

Comment:
That makes sense to me. At the moment I am using it in cca_test.go, which is in package stat, same goes for the car data of car_data_test.go. Does that mean I should add a data_test.go or something introducing aliases for stat.BostonData to avoid this confusing? I don't see an alternative unless I remove the tests that use it.